operations/mimir/ingest-storage-ingester-autoscaling.libsonnet

```diff
@@ -0,0 +1,203 @@
+    ingest_storage_ingester_autoscaling_min_replicas: error 'you must set ingest_storage_ingester_autoscaling_min_replicas in the _config in namespace %s' % $._config.namespace,
+    ingest_storage_ingester_autoscaling_max_replicas: error 'you must set ingest_storage_ingester_autoscaling_max_replicas in the _config in namespace %s' % $._config.namespace,
+
+    // The target number of active series per ingester.
+    ingest_storage_ingester_autoscaling_active_series_threshold: 1500000,
+
+    // How many zones ingesters have been deployed to.
+    ingest_storage_ingester_zones: 3,
+
+    // How long to wait before terminating an ingester after it has been notified about the scale down.
+    ingest_storage_ingester_downscale_delay: if 'querier.query-ingesters-within' in $.querier_args then
+      $.querier_args['querier.query-ingesters-within']
+    else
+      // The default -querier.query-ingesters-within in Mimir is 13 hours.
+      '13h',
+
+    // Allow to fine-tune which components gets deployed. This is useful when following the procedure
+    // to rollout ingesters autoscaling with no downtime.
+    ingest_storage_ingester_autoscaling_ingester_annotations_enabled: $._config.ingest_storage_ingester_autoscaling_enabled,
+    ingest_storage_ingester_autoscaling_replica_template_custom_resource_definition_enabled: $._config.ingest_storage_ingester_autoscaling_enabled,
+  },
+
+  // Validate the configuration.
+  assert !$._config.ingest_storage_ingester_autoscaling_enabled || $._config.multi_zone_ingester_enabled : 'partitions ingester autoscaling is only supported with multi-zone ingesters in namespace %s' % $._config.namespace,
+  assert !$._config.ingest_storage_ingester_autoscaling_ingester_annotations_enabled || $._config.multi_zone_ingester_replicas <= 0 : 'partitions ingester autoscaling and multi_zone_ingester_replicas are mutually exclusive in namespace %s' % $._config.namespace,
+  assert !$._config.ingest_storage_ingester_autoscaling_ingester_annotations_enabled || !$._config.ingester_automated_downscale_enabled : 'partitions ingester autoscaling and ingester automated downscale config are mutually exclusive in namespace %s' % $._config.namespace,
+  assert !$._config.ingest_storage_ingester_autoscaling_enabled || $.rollout_operator_deployment != null : 'partitions ingester autoscaling requires rollout-operator in namespace %s' % $._config.namespace,
+
+  //
+  // ReplicaTemplate
+  //
+
+  replica_template:: std.parseYaml(importstr 'replica-templates.yaml'),
+  replica_template_custom_resource: if !$._config.ingest_storage_ingester_autoscaling_replica_template_custom_resource_definition_enabled then null else $.replica_template,
+
+  replicaTemplate(name):: {
+    apiVersion: 'rollout-operator.grafana.com/v1',
+    kind: 'ReplicaTemplate',
+    metadata: {
+      name: name,
+      namespace: $._config.namespace,
+    },
+    spec: {
+      replicas:: null,  // We don't want this field to be managed by Flux.
```

Comment:
```suggestion
      replicas:: null,  // Hide replicas field.
```

Reply:
Done in https://github.com/grafana/mimir/pull/8028/commits/b02e79db8450f4dd7df3346097aed6528248fbfa